ethclient/ethclient.go

```diff
@@ -158,6 +158,12 @@ func (ec *Client) getBlock(ctx context.Context, method string, args ...interface
 	if err := json.Unmarshal(raw, &body); err != nil {
 		return nil, err
 	}
+	// Pending blocks don't return a block hash, compute it for sender caching.
+	if body.Hash == nil {
+		tmp := head.Hash()
+		body.Hash = &tmp
+	}
+
```

Comment:
Mostly added this addition here. This allows us to correctly set the sender caching. It seems like a nice thing to have, since we're returning a `types.Block` anyways the user can compute the block hash and it's possible they later call `TransactionSender` with it.